### PR TITLE
Fix#1862:LocalFolder Checkbox Fix

### LINKED ISSUE #1862: Security Activity issue
**Actual Behaviour**

The local folder toggle button is set to true, even when no folder is selected and the app crashes on opening the folders

**Expected Behaviour**

It shouldn't change and the app shouldn't crash.

**Steps to reproduce it**

Go to the Settings, follow the security use any password setup.
And then click on local folders, then don't select any folder nd press back the toggle button is still set to true.

**Would you like to work on the issue?**

Yes!

### PR DESCRIPTION
**Pull request by codedsun**
Fixed #1862

Changes: 
1.Refractored the layout
2.Added on back pressed listener whose absence was creating the app crash.

### CODE REVIEW

**Comment by codedsun** [comment]
@isuruAb Check and merge!

**Comment by abishekvashok** [comment]
@codedsun pls resolve conflicts.

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java [hunk 1]
```diff
@@ -131,8 +131,8 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                     title.setText("Choose folders to secure");
                     RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.secure_folder_recyclerview);
                     recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
-                    SecureDialogAdapter s = new SecureDialogAdapter();
-                    recyclerView.setAdapter(s);
+                    SecureDialogAdapter securedLocalFolders = new SecureDialogAdapter();
```
Comment (abishekvashok) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: If possible make this final.

Comment (raj10071997) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: I think there is no need to make it final as the adapter is not changing after the initialization.

Comment (abishekvashok) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: It speeds up execution if you do so.
https://developer.android.com/training/articles/perf-tips.html

Comment (raj10071997) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: Yes, you are right, declaring final does improve performance. But in local variable which is declared and accessed only once, that difference in performance will be negligible in this case.

Comment (abishekvashok) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: 1 maybe negligible compared to 1000. But 1+1+1+1+1+1... (100 times may not). It was just a suggestion though. I am okay to merge this at all costs :)

Comment (raj10071997) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: Well it was a nice suggestion even I learned something.

Comment (abishekvashok) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: Ping @codedsun

**Comment by codedsun** [comment]
@abishekvashok Merge this!

**Comment by codedsun** [comment]
@abishekvashok Please approve the PR!

**Pull request merged**